### PR TITLE
Add telemetry to our builds

### PR DESCRIPTION
**Pull request by Jeffail**
I've added documentation that I hope answers most questions users will have about this: https://github.com/redpanda-data/connect/blob/6acb0080fca75d23463c68f2e079c2c35cf30787/internal/telemetry/README.md, but ultimately some will need to look through the code themselves, so I've tried to make that as tidy and reader friendly as I can.

The telemetry only runs if the file `internal/telemetry/key.pem` is overwritten with a valid encryption key at build time, this is only done for the standard suite of official connect builds.

### CODE REVIEW

File: internal/telemetry/telemetry.go [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+	}
+
+	exportHost := defaultExportHost
+	if ExportHost != "" {
+		exportHost = ExportHost
+	}
+
+	tExporter := &telemetryExporter{
+		logger: logger,
+		Resty: resty.New().
```
Comment (mihaitodor) [inline internal/telemetry/telemetry.go:129]: That's pretty cool! I should give this library a spin.

**Review by rockwotj — APPROVED** [review]

**Review by mihaitodor — APPROVED** [review]
LGTM :shipit: Small nit: Do we want to log something and / or have some HTTP endpoint which indicates the telemetry status?

File: internal/telemetry/README.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
+Telemetry
+=========
+
+## What is this for?
+
+Our main goal is to find out the frequency with which each plugin is used in production environments, as this helps us prioritise enhancements and bugs for various plugin families on our roadmap.
```
[the remark this answers is not in the document]
Comment (Jeffail) [reply internal/telemetry/README.md:6]: I RESERVE THE RIGHT TO ADD BUGS






But yeah I'll get that fixed

[the remark this answers is not in the document]
Comment (rockwotj) [reply internal/telemetry/README.md]: > I reserve the right to add bugs

New license text right there

**Pull request merged**